klog.go

```diff
@@ -1265,6 +1265,54 @@ func Infof(format string, args ...interface{}) {
 	logging.printf(infoLog, logging.logr, format, args...)
 }
 
+// InfoS structured logs to the INFO log.
+func InfoS(msg string, keysAndValues ...interface{}) {
+	b := &bytes.Buffer{}
+	kvMap, _ := kvList2Map(keysAndValues)
+	if kvMap == nil {
+		return
+	}
+
+	b.WriteString(msg)
+
+	for key, val := range kvMap {
+		appendKeyValue(b, key, val)
+	}
+
+	logging.println(infoLog, logging.logr, b)
+}
+
+func appendKeyValue(b *bytes.Buffer, key string, value interface{}) {
+	if b.Len() > 0 {
```

Comment:
This code will never run as buffer should not be empty

Reply:
yes,  so add ' ' before fields

Reply:
My question is why we need to check b.Len() at all?

Reply:
> Please add InfoS and ErrorS methods to `Verbose` struct to allow calls like:
> 
> ```
> klog.V(2).InfoS("test", "pod", "kube-dns")
> ```

Verbose.logr is InfoLogger, it's ok to add an ErrorS?